GenderPayGap.WebUI/Controllers/Report/ReportOverviewController.cs

```diff
@@ -0,0 +1,195 @@
+﻿using System.Linq;
+using GenderPayGap.Core;
+using GenderPayGap.Core.Interfaces;
+using GenderPayGap.Database;
+using GenderPayGap.Database.Models;
+using GenderPayGap.WebUI.Helpers;
+using GenderPayGap.WebUI.Models.Report;
+using GenderPayGap.WebUI.Services;
+using GenderPayGap.WebUI.Views.Components.TaskList;
+using Microsoft.AspNetCore.Authorization;
+using Microsoft.AspNetCore.Mvc;
+
+namespace GenderPayGap.WebUI.Controllers.Report
+{
+    [Authorize(Roles = LoginRoles.GpgEmployer)]
+    [Route("account/organisations")]
+    public class ReportOverviewController : Controller
+    {
+
+        private readonly IDataRepository dataRepository;
+        private readonly DraftReturnService draftReturnService;
+
+        public ReportOverviewController(
+            IDataRepository dataRepository,
+            DraftReturnService draftReturnService)
+        {
+            this.dataRepository = dataRepository;
+            this.draftReturnService = draftReturnService;
+        }
+
+
+        [HttpGet("{encryptedOrganisationId}/reporting-year-{reportingYear}/report")]

```

Comment:
The current URLs are:
```
         organisation/{encryptedOrganisationId}/reporting-year/{reportingYear}/change-scope
account/organisations/{encryptedOrganisationId}/reporting-year-{reportingYear}/report
```

I quite like it (although this is just a personal preference thing) if each "folder" in the URLs is a valid page.
I'm hoping that, in the end, we'll have these pages:
* `account` - a "your account" homepage, with links to "manage organisations" and "manage account"
* `account/organisations` - the "manage organisations" page
* `account/organisations/{encryptedOrganisationId}` the page for an individual organisation
* `account/organisations/{encryptedOrganisationId}/reporting-year-{reportingYear}` - a page for each year (to make the organisation page a bit less visually dense)
I think `reporting-year-2019` (or maybe even `reporting-year-2019-20`) is clearer than just `2019` and gives the user a bit more information
* `account/organisations/{encryptedOrganisationId}/reporting-year-{reportingYear}/report`
`account/organisations/{encryptedOrganisationId}/reporting-year-{reportingYear}/change-scope` etc.

So `account/organisations/{encryptedOrganisationId}/reporting-year-{reportingYear}`
or  `account/organisations/{encryptedOrganisationId}/{reportingYear}`
could be valid pages (if we had a page for each year)
But `account/organisations/{encryptedOrganisationId}/reporting-year/` probably wouldn't be a valid page (because the list of reporting years will be on `account/organisations/{encryptedOrganisationId}`

What do you think?
I've probably over-though this a bit! and I'm not confident that we'll get round to creating all these pages!

Reply:
Ah, I see what you mean! Yeah I see the thinking behind that - you shouldn't really be able to remove the end off a URL and end up with a page not found, so that makes sense to me!